File: frontend/src/utils/testApi.ts
import { dashboardAPI } from '../services/api';

const testDashboardAPI = async () => {
  try {
    console.log('Testing dashboard API...');
    const data = await dashboardAPI.getDashboardData();
    console.log('Dashboard data:', data);
    console.log('Sensor readings:', data.sensorReadings?.length || 0);
    console.log('Pollution zones:', data.pollutionZones?.length || 0);
    console.log('Vehicles:', data.vehicles?.length || 0);
    console.log('Predictions:', data.predictions?.length || 0);
    
    if (data.pollutionZones?.length > 0) {
      console.log('Sample pollution zone:', data.pollutionZones[0]);
    }
    
    return data;
  } catch (error) {
    console.error('Dashboard API error:', error);
    throw error;
  }
};

// Add to window for testing in browser console
(window as any).testDashboardAPI = testDashboardAPI;

export { testDashboardAPI };

File: backend/test_api.py
#!/usr/bin/env python
import os
import django
import requests
import json

# Setup Django
os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'pollution_monitor.settings')
django.setup()

from monitoring.models import SensorReading, PollutionZone
from vehicles.models import Vehicle
from ml_prediction.models import PollutionPrediction

def test_sample_data():
    print("=== Testing Sample Data Generation ===")
    
    # Check current data
    print(f"Sensor readings: {SensorReading.objects.count()}")
    print(f"Pollution zones: {SensorReading.objects.count()}")
    print(f"Vehicles: {Vehicle.objects.count()}")
    print(f"Predictions: {PollutionPrediction.objects.count()}")
    
    # Check first few readings
    if SensorReading.objects.exists():
        print("\nFirst 3 sensor readings:")
        for reading in SensorReading.objects.all()[:3]:
            print(f"  - Location: {reading.latitude}, {reading.longitude}, AQI: {reading.aqi}")
    
    # Check zones
    if PollutionZone.objects.exists():
        print("\nPollution zones:")
        for zone in PollutionZone.objects.all():
            print(f"  - {zone.name}: {zone.zone_type} at {zone.coordinates}")
    
    # Check vehicles
    if Vehicle.objects.exists():
        print("\nVehicles:")
        for vehicle in Vehicle.objects.all()[:3]:
            print(f"  - {vehicle.name}: {vehicle.status} at {vehicle.current_latitude}, {vehicle.current_longitude}")

if __name__ == "__main__":
    test_sample_data()
